torch/distributed/distributed_c10d.py

```diff
@@ -4818,7 +4826,8 @@ def _new_group_with_tag(
     global _world
 
     default_pg = _get_default_group()
-    device_id = default_pg.bound_device_id
+    if device_id is None:
+        device_id = default_pg.bound_device_id
```

Comment:
the current logic seems fine. Shall we also do a check here to make sure the given `device_id` is the same as `default_pg.bound_device_id` if they are both not `None`?